File: auth-service/seed.js
const mongoose = require('mongoose');
const bcrypt = require('bcryptjs');
const User = require('./models/User');
require('dotenv').config();

// Sample user data
const users = [
  {
    name: 'Admin User',
    email: 'admin@example.com',
    password: 'admin123',
    role: 'admin'
  },
  {
    name: 'Regular User',
    email: 'user@example.com',
    password: 'user123',
    role: 'user'
  }
];

// Connect to MongoDB
const seedDatabase = async () => {
  try {
    await mongoose.connect(process.env.MONGODB_URI);
    console.log('Connected to MongoDB');

    // Delete existing users
    await User.deleteMany({});
    console.log('Deleted existing users');

    // Create users directly and let the pre-save hook handle password hashing
    for (const userData of users) {
      const user = new User({
        name: userData.name,
        email: userData.email,
        password: userData.password,
        role: userData.role
      });

      await user.save();
    }

    console.log('Added sample users to database');
    console.log('Database seeding completed successfully');

    // Print login credentials for reference
    console.log('\nYou can now log in with these credentials:');
    console.log('Admin: admin@example.com / admin123');
    console.log('User: user@example.com / user123');

    process.exit(0);
  } catch (error) {
    console.error('Error seeding database:', error.message);
    process.exit(1);
  }
};

// Run the seed function
seedDatabase();

File: auth-service-python/seed.py
from models.user import User
from config.db import Database

# Sample user data
users = [
    {
        'name': 'Admin User',
        'email': 'admin@example.com',
        'password': 'admin123',
        'role': 'admin'
    },
    {
        'name': 'Regular User',
        'email': 'user@example.com',
        'password': 'user123',
        'role': 'user'
    }
]

def seed_database():
    try:
        # Initialize database connection
        db = Database.get_instance().get_db()
        
        # Delete existing users
        db.users.delete_many({})
        print('Deleted existing users')
        
        # Create users
        for user_data in users:
            user = User(
                name=user_data['name'],
                email=user_data['email'],
                password=user_data['password'],
                role=user_data['role']
            )
            user.save()
            print(f"Added user: {user.email}")
        
        print('\nDatabase seeding completed successfully')
        print('\nYou can now log in with these credentials:')
        print('Admin: admin@example.com / admin123')
        print('User: user@example.com / user123')
        
    except Exception as e:
        print(f"Error seeding database: {str(e)}")
    finally:
        # Close database connection
        Database.get_instance().close()

if __name__ == '__main__':
    seed_database()
